src/tbb/arena.cpp

```diff
@@ -232,14 +232,13 @@ void arena::free_arena () {
     __TBB_ASSERT( !my_global_concurrency_mode, NULL );
 #endif
     poison_value( my_guard );
-    std::intptr_t drained = 0;
     for ( unsigned i = 0; i < my_num_slots; ++i ) {
         // __TBB_ASSERT( !my_slots[i].my_scheduler, "arena slot is not empty" );
         // TODO: understand the assertion and modify
         // __TBB_ASSERT( my_slots[i].task_pool == EmptyTaskPool, NULL );
         __TBB_ASSERT( my_slots[i].head == my_slots[i].tail, NULL ); // TODO: replace by is_quiescent_local_task_pool_empty
         my_slots[i].free_task_pool();
-        drained += mailbox(i).drain();
+        static_cast<void>(mailbox(i).drain());
```

Comment:
I meant changing the function signature in [mailbox.h:189-197](https://github.com/oneapi-src/oneTBB/blob/master/src/tbb/mailbox.h#L189-L197):
```diff
-intptr_t drain() {
-    intptr_t k = 0;
-    // No fences here because other threads have already quit.
-    for( ; task_proxy* t = my_first; ++k ) {
-        my_first.store(t->next_in_mailbox, std::memory_order_relaxed);
-        // cache_aligned_deallocate((char*)t - task_prefix_reservation_size);
-    }
-    return k;
-}
+void drain() {
+    // No fences here because other threads have already quit.
+    for( ; task_proxy* t = my_first; ) {
+        my_first.store(t->next_in_mailbox, std::memory_order_relaxed);
+        // cache_aligned_deallocate((char*)t - task_prefix_reservation_size);
+    }
+}
```